calc/l3_route_resolver.go

```diff
@@ -525,11 +536,20 @@ func (c *L3RouteResolver) flush() {
 					rt.Type = proto.RouteType_REMOTE_HOST
 				}
 			}
-			if ri.LocalWEP.RefCount > 0 {
-				// We have a local WEP with this IP.
-				rt.DstNodeName = c.myNodeName
-				rt.Type = proto.RouteType_LOCAL_WORKLOAD
-				rt.LocalWorkload = true
+
+			for nodename, _ := range ri.WEP.RefCount {
```

Comment:
Map iteration is not deterministic (it's randomised each time).  I think it's really important to keep the calc graph deterministic for a few reasons:

- It really helps when reasoning about it (e.g. when diagnosing an issue hit in the field)
- It's part of the contract that other parts of felix expect (I don't want to have to revisit other parts of felix to check for invalidated assumptions if we allow a bit of non-determinism).
- It's expected and asserted by the calc graph FV tests (which you should use to test this new feature).   If you add a test in that framework that has two overlapping WEPs, it should detect the non-determinism

One way to quickly get determinism here is to use a tie-break rule such as "highest nodename wins".